pkg/domain/filters/volumes.go

```diff
@@ -17,7 +18,11 @@ func GenerateVolumeFilters(filters url.Values) ([]libpod.VolumeFilter, error) {
 			case "name":
 				nameVal := val
 				vf = append(vf, func(v *libpod.Volume) bool {
-					return nameVal == v.Name()
+					match, err := regexp.MatchString(nameVal, v.Name())
+					if err != nil {
+						return false
+					}
+					return match
```

Comment:
I think it would be faster to use `regex.Compile()` to compile the expression into a new `Regexp` object outside of the new anonymous filter function.  This would also give us the opportunity to flag any syntax errors while we're still in `GenerateVolumeFilters()`.

The anonymous filter function that's created here would then call that `Regexp` object's `MatchString()` method instead of calling the regexp package's `MatchString()` function, which I expect has to compile the regex every time it's called.

It looks like `libpod.Runtime.Volumes()` will call each of these filter functions for every volume, and that's got to add up.

Reply:
I thought about it, the problem I still need to pass the pattern which is available only inside the scope of `GenerateVolumeFilters()` - `nameVal := val` which this will be `Compile`d everytime `GenerateVolumeFilters()` is called. Unless, I can "cache" this in a map.

Or maybe I am missing something. Lemme know. Thanks.

Reply:
The anonymous function (I looked it up and it turns out I should have used the phrase "function literal") can access variables defined in `GenerateVolumeFilters()` so long as the function itself is within the scope where the variable exists, as described at https://go.dev/ref/spec#Function_literals.

Reply:
Oh right! This will work because `vfs` holds anonymous functions. Yes, that'll work. Thanks for the suggestions.